src/components/TodoItem.js

```diff
@@ -5,32 +5,38 @@ import ListItemText from '@mui/material/ListItemText';
 import Checkbox from '@mui/material/Checkbox';
 import IconButton from '@mui/material/IconButton';
 import DeleteIcon from '@mui/icons-material/Delete';
+import useTodoStore from "../stores/useTodoStore";
 
 export function TodoItem(params) {
-  const {text, completed} = params;
-  const labelId = `checkbox-list-label-${text}`;
 
-  return (
-    <ListItem
-      secondaryAction={
-        <IconButton edge="end" aria-label="deleteTodo">
-          <DeleteIcon/>
-        </IconButton>
-      }
-      disablePadding
-    >
-      <ListItemButton role={undefined} dense>
-        <ListItemIcon>
-          <Checkbox
-            edge="start"
-            checked={completed}
-            tabIndex={-1}
-            disableRipple
-            inputProps={{ 'aria-labelledby': labelId }}
-          />
-        </ListItemIcon>
-        <ListItemText id={labelId} primary={text} />
-      </ListItemButton>
-    </ListItem>
-  )
-}
\ No newline at end of file
+    const {text, completed, index} = params;
+    const labelId = `checkbox-list-label-${text}`;
+
+    const deleteTodo = useTodoStore((state) => state.delete);
+    const toggleTodo = useTodoStore((state) => state.toggle);
+
+    return (
+        <ListItem
+            secondaryAction={
+                <IconButton edge="end" aria-label="deleteTodo" onClick={(event) => {deleteTodo(index)}}>
+                    <DeleteIcon/>
+                </IconButton>
+            }
+            disablePadding
+        >
+            <ListItemButton role={undefined} dense>
+                <ListItemIcon>
+                    <Checkbox
+                        edge="start"
+                        checked={completed}
+                        tabIndex={-1}
+                        disableRipple
+                        inputProps={{'aria-labelledby': labelId}}
+                        onChange={(event) => {toggleTodo(index)}}
```

Comment:
if the `event` value is not used in the function, we can delete it(it reduce the complexity of code)

```suggestion
                        onChange={() => {toggleTodo(index)}}
```